velox/type/Type.h

```diff
@@ -455,6 +455,16 @@ class Type : public Tree<const TypePtr>, public velox::ISerializable {
 
   virtual bool isPrimitiveType() const = 0;
 
+  /// Return true if the type is scalar type or unknown type. Return false if
```

Comment:
This is a virtual method and therefore it cannot say what its implementations will return. Instead in need to explain what the return value means. How about,

```
Returns true if equality relationship is defined for the values of this type, i.e. a == b is defined and returns true, false or null.

For example, scalar types are usually comparable and complex types are comparable if their nested types are.
```

Reply:
Got it, updated.